src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -647,17 +686,29 @@ public function bulkAction(Request $request, $action)
      */
     public function massEditAction(Request $request, $action)
     {
-        $productProvider = $this->container->get('prestashop.core.admin.data_provider.product_interface');
+        $translator = $this->get('translator');
+
+        if (!$this->isGranted(ProductVoter::UPDATE)) {
+            $errorMessage = $translator->trans(
+                'You do not have permission to modify this.',
```

Comment:
Could it be "You do not have permission to edit this."?
(to avoid creating a new string)
Thanks!